File: new_script/{{cookiecutter.project_slug}}/template/argv.py
#!/usr/bin/env python3
#
# Written by {{ cookiecutter.name }}
# Year {{ cookiecutter.year }}
#

import argparse
import os.path
import fileinput
ver = "0.1.0"
parser = argparse.ArgumentParser()

#
# positional arguments
#

# default type is string
parser.add_argument(
        "infile",
        help = "STDOUT from AbRSA",
)
# specify integer type
parser.add_argument(
        "number",
        help = "display a number",
        type = int
)

#
# optional arguments
#

# short and long options
# store True if specified
parser.add_argument(
        "-b",
        "--verbose",
        help = "verbose mode",
        action = "store_true"
)
parser.add_argument(
        "-v",
        "--version",
        help = "show script version and exit",
        action = "version",
        version = ver
)

# set choices for argument and default value
parser.add_argument(
        "-p",
        "--threads",
        help = "number of threads",
        choices = range(1,9),
        default = 2,
        type = int
)

args = parser.parse_args()

if args.verbose:
    print("Verbose mode")

if args.threads:
    print("Using %d threads" % args.threads)

# stream from STDIN
if args.infile == "-":
    for line in fileinput.input():
        line = line.rstrip()
        print(line)
else:
    if os.path.exists(args.infile):
        with open(args.infile) as f:
            for line in f:
                line = line.rstrip()
                print(line)
    else:
        print(f"{args.infile} does not exist")
        quit(1)

File: new_script/{{cookiecutter.project_slug}}/template/argv.R
#!/usr/bin/env Rscript
#
# Written by {{ cookiecutter.name }}
# Year {{ cookiecutter.year }}
#

script_ver <- '0.1.0'

suppressPackageStartupMessages(library(optparse))

# optional args
option_list <- list(
   make_option(c("-p", "--threads"), type = "integer", default = 2, help =
               "Number of threads to use (default = %default)"),
   make_option(c("-v", "--version"), action = "store_true", help =
               "Show script version and exit"),
   make_option(c("-b", "--verbose"), action = "store_true", default = FALSE,
               help = "Verbose mode (default = %default)")
)

# create your own usage
opt_parse <- OptionParser(usage = "%prog [options] infile <infile2> ...",
                          option_list = option_list)

# set positional_arguments to TRUE
opt <- parse_args(opt_parse, positional_arguments = TRUE)

if(length(opt$options$version) > 0){
   message(script_ver)
   quit()
}

# print usage if no positional args provided
if (length(opt$args) == 0){
   print_help(opt_parse)
   stop("Please provide an input file")
}

if (opt$options$verbose){
   message("Verbose mode activated")
   message(paste0("Using ", opt$options$threads, " threads"))
}

for (file in opt$args){
   if(!file.exists(file)){
      stop(paste0(file, " does not exist"))
   }
   message(paste0("Processing ", file))
}

message("Done")
quit()
